tka/tka.go

```diff
@@ -160,12 +164,18 @@ func advanceChain(state State, candidates []AUM) (next *AUM, out State, err erro
 	return &aum, state, nil
 }
 
-// fastForward iteratively advances the current state based on known AUMs until
-// the given termination function returns true or there is no more progress possible.
+// fastForwardWithAdvancer iteratively advances the current state by calling
+// the given advancer to get+apply the next update. This process is repeated
+// until the given termination function returns true or there is no more
+// progress possible.
 //
 // The last-processed AUM, and the state computed after applying the last AUM,
 // are returned.
-func fastForward(storage Chonk, maxIter int, startState State, done func(curAUM AUM, curState State) bool) (AUM, State, error) {
+func fastForwardWithAdvancer(
+	storage Chonk, maxIter int, startState State,
+	advancer func(state State, candidates []AUM) (next *AUM, out State, err error),
```

Comment:
I see how you got here, but this is really the outer limit of what is practical with passing closures around before it becomes unreadable. :)

You're also pretty close to being able to combine advancer and done, but we can think about that another time.

Reply:
Yeah definitely XD If we end up needing to further complicate fast forward, i think its time for another abstraction.